scripts/quests/sandoria/DRG_AF1_A_Craftmans_Work.lua

```diff
@@ -0,0 +1,133 @@
+                        return quest:progressEvent(71)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [73] = function(player, csid, option, npc)
+                    if option == 1 then
+                        quest:begin(player)
+                        quest:setVar(player, 'Prog', 1)
+                    else
+                        quest:setVar(player, 'csSeen', 1) -- Declined First Offering.
+                    end
+                end,
+
+                [71] = function(player, csid, option, npc)
+                    if option == 1 then -- Accept quest.
+                        quest:begin(player)
+                        quest:setVar(player, 'Prog', 1)
+                    end
+                end,
+            },
+        },
+    },
+
+    -- Section: Quest accepted
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.NORTHERN_SAN_DORIA] =
+        {
+            ['Miaux'] =
+            {
+                onTrigger = function(player, npc)
+                    if player:hasKeyItem(xi.ki.ALTEPA_POLISHING_STONE) then
+                        return quest:progressEvent(70)
+                    else
+                        return quest:event(69)
+                    end
+                end,
+            },
+
+        onEventFinish =
+            {
+                [70] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:delKeyItem(xi.ki.ALTEPA_POLISHING_STONE)
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.EASTERN_ALTEPA_DESERT] =
+        {
+            ['qm'] =
+            {
+                onTrigger = function(player, npc)
+                    if
+                        quest:getVar(player, 'Prog') == 1 and
+                        quest:getVar(player, 'nmKilled') == 0
+                    then
+                        if npcUtil.popFromQM(player, npc, easternAltepaID.mob.DECURIO_I_III, { claim = true, hide = 0 }) then
+                            player:messageSpecial(easternAltepaID.text.HOSTILE_GAZE)
+                        end
+                    elseif quest:getVar(player, 'nmKilled') == 1 then
+                        npcUtil.giveKeyItem(player, xi.ki.ALTEPA_POLISHING_STONE)
+                        quest:setVar(player, 'nmKilled', 0)
+                        quest:setVar(player, 'Prog', 2)
+                        return quest:noAction()
+                    end
+                end,
+            },
+
+            ['Decurio_I-III'] =
+            {
+                onMobDeath = function(mob, player, optParams)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        quest:setVar(player, 'nmKilled', 1)
+                    end
```

Comment:
So, we are using 3 variables in this quest. We could get away with just using 1 to track the whole progression.

We have to think that players leave quests unfinished. Long term, the less variables used = less variables left living in the database.

So it's more of a "good practice" thing, than a matter of it being wrong.

Reply:
Like changing it to use setBit()

Reply:
Understood - i went back and forth on trimming the fat but again, opted to go with what previous vars were set at on the conversion instead of a complete rewrite.